git/github/github.go

```diff
@@ -302,6 +318,89 @@ func (c *Client) GetCheckRunID(ctx context.Context, owner string, repo string, S
 	return nil, nil
 }
 
+// GetAllCheckRunsForRef returns all existing GitHub CheckRuns if a match for the Owner, Repo, SHA, and appID.
+func (c *Client) GetAllCheckRunsForRef(ctx context.Context, owner string, repo string, SHA string, appID int64) ([]*ghapi.CheckRun, error) {
+	filter := "all"
+
+	res, _, err := c.GetChecksService().ListCheckRunsForRef(
+		ctx,
+		owner,
+		repo,
+		SHA,
+		&ghapi.ListCheckRunsOptions{
+			AppID:  &appID,
+			Filter: &filter,
+		},
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if *res.Total == 0 {
+		c.logger.Info("Found no CheckRuns for the ref", "SHA", SHA)
+		return nil, nil
+	}
+
+	return res.CheckRuns, nil
+}
+
+// GetExistingCheckRun returns existing GitHub CheckRun ExternalID in cra.
+func (c *Client) GetExistingCheckRun(checkRuns []*ghapi.CheckRun, newCheckRun *CheckRunAdapter) *ghapi.CheckRun {
+	for _, cr := range checkRuns {
+		cr := cr
```

Comment:
What is the purpose of this declaration?

Reply:
Thanks, we don't need it since not using `&`. Removed.